### PR TITLE
Add tests for build, lint and format

### PR DESCRIPTION
**Pull request by PascalHelbig**

### CODE REVIEW

File: tests/build.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const { execSync } = require('child_process');
+
+test('build runs without an error', () =>
+  expect(() =>
+    execSync('node ../bin build', {
+      cwd: 'tests',
```
Comment (cangoektas) [inline tests/build.test.js:6]: I think I liked the calls to `execSync` with `node bin` more. Let's maybe change this to stay consistent with the other tests :)

File: tests/src/index.js [hunk 2]
```diff
@@ -0,0 +1 @@
+console.log('Hello World');
```
Comment (cangoektas) [inline tests/src/index.js]: I would also add the syntax features that the build script should support. I think the following should be enough:

```js
console.log(
  'Hello',
  'World', // traling function comma
);

class A {
  foo = 'bar'; // class properties
}

async function shouldBeGenerator() {
  return null;
}
```

Ideally, the tests would not only test that the build script doesn't fail but also that the transpiled output is correct. Don't know how exactly now but we can do that in the next iteration!

**Review by cangoektas — COMMENTED** [review]
This is really really cool! I was thinking about how to test this without worrying too much about the filesystem but testing with the `--check` gets around that issue. 😄

File: tests/build.test.js (hunk 1 above)
Comment (PascalHelbig) [reply tests/build.test.js:6]: `build` looks for a `src` folder. Therefore the working directory has to be the `test` directory. :confused:

Comment (cangoektas) [reply tests/build.test.js:6]: Ahhh... right... we don't have support for targets in the build script. You're right!

**Comment by cangoektas** [comment]
Awesome, thank you ❤️

**Pull request merged**